generate_status_quo.py

```diff
@@ -0,0 +1,345 @@
+# Ideally, -you- munge this script until no files are orphaned.
+
+
+import argparse
+import copy
+import os
+import shutil
+import subprocess
+
+from collections import OrderedDict
+from logzero import logger
+import ruamel.yaml
+from sh import git
+from sh import find
+
+
+
+def run_command(cmd):
+    p = subprocess.Popen(cmd, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    (so, se) = p.communicate()
+    return (p.returncode, so, se)
+
+
+class StatusQuo:
+
+    collections = None
+    plugins = None
+    pluginfiles = None
+    orphaned = None
+    topics = None
+
+    synonyms = {
+        'apache-libcloud': 'cloud.misc',
+        'aws': 'amazon',
+        #'azure_rm': 'azure',
+        'bigip': 'f5',
+        'bigiq': 'f5',
+        'ce': 'cloudengine',
+        'checkpoint': 'check_point',
+        'cloudforms': 'cloud',
+        #'cloudstack': 'cloud.cloudstack',
+        #'consul': 'clustering.consul',
+        #'dns': 'net_tools',
+        'ec2': 'amazon',
+        #'ecs': 'crypto.entrust',
+        #'kube': 'k8s',
+        #'foreman': 'remote_management.foreman',
+        'spacewalk': 'remote_management',
+        'gce': 'google',
+        'gcp': 'google',
+        #'hashi_vault': 'crypto',
+        'hetzner': 'hcloud',
+        'hwc': 'huawei',
+        #'jabber': 'web_infrastructure',
+        'infinibox': 'infinidat',
+        #'infoblox': 'net_tools.nios',
+        'infoblox': 'nios',
+        #'ipa': 'identity.ipa',
+        'kubectl': 'k8s',
+        'libcloud': 'cloud.misc',
+        #'linode': 'cloud.linode',
+        #'lxc': 'cloud.lxc',
+        #'lxd': 'cloud.lxc',
+
+        #'mso': 'network.aci',
+        'mso': 'aci',
+        #'nagios': 'monitoring.nagios',
+        #'nagios': 'web_infrastructure',
+        'oc': 'k8s',
+        #'openstack': 'cloud.openstack',
+        #'openshift': 'clustering.openshift',
+        #'openvz': 'cloud.misc',
+        #'ovirt': 'cloud.ovirt',
+        #'onepassword': 'identity',
+
+        #'package': 'packaging',
+        'package': 'packaging.os',
+        'passwordstore': 'identity',
+
+        'powershell': 'windows',
+        #'proxmox': 'cloud.misc',
+        'psrp': 'windows',
+        'rax': 'cloud.rackspace',
+        #'redis': 'database.misc',
+        'rhv': 'ovirt',
+        'tower': 'ansible_tower',
+        #'utm': 'sophos_utm',
+        'vagrant': 'cloud',
+        'vca': 'vmware',
+        #'virtualbox': 'cloud.misc',
+        #'vbox': 'cloud.misc',
+
+        'win': 'windows',
+        'win': 'windows',
+
+        'Vmware': 'vmware',
+        'vmware': 'cloud.vmware',
+        #'yum': 'packaging.os',
+        #'zabbix': 'monitoring.zabbix'
+    }
+
+    def __init__(self):
+        self.collections = OrderedDict()
+        self.url = 'https://github.com/ansible/ansible'
+        self.checkouts_dir = '.cache/checkouts'
+        self.checkout_dir = os.path.join(self.checkouts_dir, 'ansible')
+
+    def run(self):
```

Comment:
This is a distinct use case for an alternate constructor:
```suggestion
    @classmethod
    def run(cls):
        self = cls()
        self.manage_checkout()
        self.get_plugins()
        self.make_spec()
```

Then, you could just invoke it entry-point style, one-step invocation instead of a multi-step one:
```python
StatusQuo.run()
```